x-pack/plugins/enterprise_search/public/applications/app_search/components/result/result.tsx

```diff
@@ -33,6 +43,19 @@ export const Result: React.FC<Props> = ({ result, showScore }) => {
     [result]
   );
   const numResults = resultFields.length;
+  const typeForField = (fieldName: string) => {
+    if (schemaForTypeHighlights) return schemaForTypeHighlights[fieldName];
+  };
+
+  const conditionallyLinkedArticle = (children: React.ReactNode) => {
+    if (!shouldLinkToDetailPage)
+      return <article className="appSearchResult__content">{children}</article>;
+    return (
+      <ReactRouterHelper to={getDocumentDetailRoute(resultMeta.engine, resultMeta.id)}>
+        <a className="appSearchResult__content">{children}</a>
+      </ReactRouterHelper>
+    );
```

Comment:
Minor formatting suggestion, I think the indenting a ternary return makes it a little easier to read here (plus I generally somewhat prefer positive if statements over negative)

```suggestion
    return shouldLinkToDetailPage ? (
      <ReactRouterHelper to={getDocumentDetailRoute(resultMeta.engine, resultMeta.id)}>
        <a className="appSearchResult__content">{children}</a>
      </ReactRouterHelper>
    ) : (
      <article className="appSearchResult__content">{children}</article>
    );
```

Reply:
ba3c0bb116f82b12018f0e983ec3a7271c9435ed

Reply:
ba3c0bb116f82b12018f0e983ec3a7271c9435ed